datalad_container/containers_run.py

```diff
@@ -0,0 +1,112 @@
+
+__docformat__ = 'restructuredtext'
+
+import logging
+from simplejson import loads
+from argparse import REMAINDER
+
+from datalad.interface.base import Interface
+from datalad.interface.base import build_doc
+from datalad.support.param import Parameter
+from datalad.distribution.dataset import datasetmethod
+from datalad.distribution.dataset import require_dataset
+from datalad.interface.utils import eval_results
+
+from datalad.interface.run import Run
+from datalad_container.containers_list import ContainersList
+
+lgr = logging.getLogger("datalad.containers.containers_run")
+
+
+_run_params = dict(
+    Run._params_,
+    container_name=Parameter(
+        args=('-n', '--container-name',),
+        metavar="NAME",
+        doc="""Specify the name of a known container to use for execution,
+        in case multiple containers are configured."""),
+    # have to override the cmd parameter, because in `run` it says
+    # SHELL COMMAND but with a container it could be something totally
+    # different
+    cmd=Parameter(
+        args=("cmd",),
+        nargs=REMAINDER,
+        metavar='COMMAND',
+        doc="command for execution"),
+)
+
+
+@build_doc
+# all commands must be derived from Interface
+class ContainersRun(Interface):
+    # first docstring line is used a short description in the cmdline help
+    # the rest is put in the verbose help and manpage
+    """Drop-in replacement of 'run' to perform containerized command execution
+
+    Container(s) need to be configured beforehand (see containers-add).
+    If only one container is known, it will be selected automatically,
+    otherwise a specific container has to be selected.
```

Comment:
minor, but I think replacing the second "selected" with "specified" would be clearer.

Reply:
Done